source/module_esolver/esolver_ks_lcao.cpp

```diff
@@ -724,10 +724,39 @@ void ESolver_KS_LCAO::eachiterfinish(int iter)
         const int precision = 3;
 
         std::stringstream ssc;
-        std::stringstream ss1;
         ssc << GlobalV::global_out_dir << "tmp"
-            << "_SPIN" << is + 1 << "_CHG";
-        ModuleIO::write_rho(pelec->charge->rho_save[is], is, iter, ssc.str(), precision); // mohan add 2007-10-17
+            << "_SPIN" << is + 1 << "_CHG.cube";
+        double ef_tmp;
+        if(is==0 && GlobalV::NSPIN==2)
```

Comment:
These code repeat at least 5 times, please add a function in GlobalC::en:
```
double ef_tmp = GlobalC::en.get_ef(is);
```

Reply:
These code block is not correct that ef_up and ef_dw only used when GlobalV::TWO_EFERMI=true:
```
double energy::get_ef(const int& is)
{
    if(!GlobalV::TWO_EFERMI)
    {
        return ef;
    }
    else if(is == 0)
    {
        return ef_up;
    }
    else if(is == 1)
    {
        return ef_dw;
    }
    else
    {
        WARNING_QUIT("","");
    }
}
```